### PR TITLE
[#3277] Add template tag to get dictionary value

### LINKED ISSUE #3277: Using a list as data source that contains items with spaces creates values that are unusable in templates
### Product versie / Product version

2.2.0

### Omschrijf het probleem / Describe the bug

When a list is used as data source for checboxes/selectboxes component the items in the list are duplicated for key + value. If these items contain spaces the values that are mapped are unusable in templates. 

### Stappen om te reproduceren / Steps to reproduce

List with 3 items with spaces
image

use this list in checkbox component:
image

looks good:
image

But using these values in templates is not possible:
image
 

### Verwacht gedrag / Expected behavior

When the item from the list are duplicated to Key & Value, perhaps the spaces should be stripped to make them usable in templates?

### Screen resolution

None

### Device

None

### OS

None

### Browser

_No response_

### PR DESCRIPTION
**Pull request by SilviaAmAm**
Fixes #3277

### CODE REVIEW

File: docs/manual/templates.rst [hunk 1]
```diff
@@ -155,6 +155,27 @@ Expressie                           Voorbeeld waarde                   Toelichti
 .. note:: Op dit moment krijgt u altijd de Nederlandse vertalingen/lokalisatie.
    Er is nog geen ondersteuning voor andere talen.
 
+Template tags
+-------------
+
+De volgende template tags kunnen ook worden gebruikt in opmaak componenten.
```
Comment (sergei-maertens) [inline docs/manual/templates.rst:161]:
```suggestion
De volgende template tags kunnen ook worden gebruikt in opmaakcomponenten.
```

File: docs/manual/templates.rst [hunk 2]
```diff
@@ -155,6 +155,27 @@ Expressie                           Voorbeeld waarde                   Toelichti
...
+
+**get_value**
+
+Geeft de mogelijkheid om een waarde uit een dictionary uit te halen.
+Bijvoorbeeld, als deze variabele in het formulier bestaat:
+
+.. code:: python
+
+   eenVariabel = {"optie 1": "waarde 1", "optie 2": "waarde 2"}
```
Comment (sergei-maertens) [inline docs/manual/templates.rst:170]:
```suggestion
   eenVariabele = {"optie 1": "waarde 1", "optie 2": "waarde 2"}
```

File: docs/manual/templates.rst [hunk 3]
```diff
@@ -155,6 +155,27 @@ Expressie                           Voorbeeld waarde                   Toelichti
...
+
+Dan de template tag ``get_value``:
```
Comment (sergei-maertens) [inline docs/manual/templates.rst:172]:
```suggestion
Dan zal de template tag ``get_value``:
```

File: docs/manual/templates.rst [hunk 4]
```diff
@@ -155,6 +155,27 @@ Expressie                           Voorbeeld waarde                   Toelichti
...
+
+.. code::
```
Comment (sergei-maertens) [inline docs/manual/templates.rst:174]:
```suggestion
.. code:: django
```
there exists syntax highlighting if I remember correctly :)

File: docs/manual/templates.rst [hunk 5]
```diff
@@ -155,6 +155,27 @@ Expressie                           Voorbeeld waarde                   Toelichti
...
+
+   {% get_value eenVariabel 'optie 1' %}
```
Comment (sergei-maertens) [inline docs/manual/templates.rst:176]:
```suggestion
   {% get_value eenVariabele 'optie 1' %}
```

File: docs/manual/templates.rst [hunk 6]
```diff
@@ -155,6 +155,27 @@ Expressie                           Voorbeeld waarde                   Toelichti
...
+
+Geeft de waarde ``waarde 1`` terug.
```
Comment (sergei-maertens) [inline docs/manual/templates.rst:178]:
```suggestion
de waarde ``waarde 1`` teruggeven.
```

**Review by sergei-maertens — COMMENTED** [review]
I admire your dedication to writing documentation in a foreign language!

File: src/openforms/utils/templatetags/utils.py [hunk 7]
```diff
@@ -0,0 +1,8 @@
+from django import template
```
Comment (sergei-maertens) [inline src/openforms/utils/templatetags/utils.py:1]: Probably wise to put a big fat disclaimer at the top that any tags and filters defined here are automatically enabled for user-submitted templates.

File: src/openforms/utils/templatetags/utils.py [hunk 8]
```diff
@@ -0,0 +1,8 @@
...
+
+register = template.Library()
+
+
+@register.simple_tag(name="get_value")
+def get_value(dict, value_key):
+    return dict.get(value_key, "")
```
Comment (sergei-maertens) [inline src/openforms/utils/templatetags/utils.py:8]:
```suggestion
def get_value(value: dict, value_key):
    return value.get(value_key, "")
```
shadowing the builtin `dict` looks a bit odd. I'd also like to see some tests for non-dict values, people are definitely going to use this the wrong way. Try this on strings, lists...

I think the behaviour on non-compatible types should be "return empty string" (which is different from the normal behaviour of "let template tags crash")

File: docs/manual/templates.rst (hunk 3 above)
Comment (SilviaAmAm) [reply docs/manual/templates.rst:172]: Shall I also change the last part `geeft de waarde 'waarde 1' terug` to `de waarde 'waarde 1' teruggeven`? And maybe also put it on the same line :thinking:

File: docs/manual/templates.rst (hunk 4 above)
Comment (SilviaAmAm) [reply docs/manual/templates.rst:174]: :exploding_head:

File: docs/manual/templates.rst (hunk 3 above)
Comment (SilviaAmAm) [reply docs/manual/templates.rst:172]: Ohh nvm you specified it below :smile:

File: docs/manual/templates.rst [hunk 9]
```diff
@@ -155,6 +155,26 @@ Expressie                           Voorbeeld waarde                   Toelichti
...
    Er is nog geen ondersteuning voor andere talen.
 
+Template tags
+-------------
+
+De volgende template tags kunnen ook worden gebruikt in opmaakcomponenten.
+
+**get_value**
+
+Geeft de mogelijkheid om een waarde uit een sleutel-waarde variabelen uit te halen.
```
Comment (sergei-maertens) [inline docs/manual/templates.rst:165]:
```suggestion
Geeft de mogelijkheid om een waarde uit een sleutel-waarde variabele te halen.
```

**Review by sergei-maertens — APPROVED** [review]
One more documentation fix, but looks good!

File: docs/manual/templates.rst (hunk 9 above)
Comment (SilviaAmAm) [reply docs/manual/templates.rst:165]: oups, modified this too fast :see_no_evil:

**Pull request merged**